lib/ups/builders/builder_base.rb

```diff
@@ -144,13 +144,25 @@ def add_ship_from(opts = {})
       # @return [void]
       def add_package(opts = {})
         shipment_root << Element.new('Package').tap do |org|
-          org << packaging_type
+          org << packaging_type(opts[:packaging_type] || customer_packaging)
           org << element_with_value('Description', 'Rate')
           org << package_weight(opts[:weight], opts[:unit])
-          org << package_dimensions(opts[:dimensions]) if opts[:dimensions]
+          org << package_dimensions(opts[:dimensions]) if dimensions_required?(opts)
         end
       end
 
+      def customer_packaging
+        {code: '02', description: 'Customer Supplied Package'}
+      end
+
+      def dimensions_required? (opts = {})
+        return unless opts[:dimensions]
+
+        return true unless opts[:packaging_type]
```

Comment:
Is the added logic needed, does UPS throw an error if dimensions are provided with a standard(non customer supplied) package?